File: 57. Insert Interval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        if not newInterval:
            return intervals
        if not intervals:
            return [newInterval]
        left, right = [], []
        for i in intervals:
            if i[1] < newInterval[0]:
                left.append(i)
            elif i[0] > newInterval[1]:
                right.append(i)
            else:
                newInterval[0] = min(newInterval[0], i[0])
                newInterval[1] = max(newInterval[1], i[1])
        return left + [newInterval] + right

File: 57. Insert Interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        if (intervals==null || intervals.length==0) {
            int[][] ans = {newInterval};
            return ans;
        }
        if (newInterval==null || newInterval.length==0) {
            return intervals;
        }
        boolean flag = true;
        LinkedList<int[]> ls = new LinkedList<>();
        for (int i=0; i<intervals.length; i++) {
            int[] temp = intervals[i];
            if (temp[1] < newInterval[0]) {
                ls.add(temp);
            } else if (temp[0] > newInterval[1]) {
                if (flag) {
                    ls.add(newInterval);
                    flag = false;
                }
                ls.add(temp);
            } else {
                newInterval[0] = Math.min(temp[0], newInterval[0]);
                newInterval[1] = Math.max(temp[1], newInterval[1]);
            }
        }
        if (flag) {
            ls.add(newInterval);
            flag = false;
        }
        return ls.toArray(new int[ls.size()][2]);
    }
}